File: mobile/lib/core/theme/joi_colors.dart
import 'dart:ui';

abstract final class JoiColors {
  // Core backgrounds
  static const background = Color(0xFF0A0A0F);
  static const surface = Color(0xFF12121A);
  static const surfaceVariant = Color(0xFF1A1A25);
  static const surfaceHigh = Color(0xFF222230);

  // Accent — Cyan
  static const primary = Color(0xFF00E5FF);
  static const primaryMuted = Color(0xFF0097A7);
  static const primaryGlow = Color(0x3300E5FF);

  // Accent — Amber
  static const secondary = Color(0xFFFFAB40);
  static const secondaryMuted = Color(0xFFC77800);
  static const secondaryGlow = Color(0x33FFAB40);

  // Accent — Violet
  static const tertiary = Color(0xFFB388FF);
  static const tertiaryMuted = Color(0xFF7C4DFF);

  // Semantic
  static const error = Color(0xFFFF5252);
  static const success = Color(0xFF69F0AE);
  static const warning = Color(0xFFFFD740);
  static const info = Color(0xFF40C4FF);

  // Text
  static const textPrimary = Color(0xFFF5F5F7);
  static const textSecondary = Color(0xFFB0B0C0);
  static const textTertiary = Color(0xFF6B6B80);
  static const textOnPrimary = Color(0xFF0A0A0F);

  // Borders & dividers
  static const border = Color(0xFF2A2A3A);
  static const borderSubtle = Color(0xFF1E1E2E);
  static const divider = Color(0xFF1A1A25);
}

File: app/JOI/Core/Theme/JOIColors.swift
import SwiftUI

extension Color {
    init(hex: UInt, opacity: Double = 1.0) {
        self.init(
            .sRGB,
            red: Double((hex >> 16) & 0xFF) / 255.0,
            green: Double((hex >> 8) & 0xFF) / 255.0,
            blue: Double(hex & 0xFF) / 255.0,
            opacity: opacity)
    }
}

enum JOIColors {
    // Core backgrounds
    static let background = Color(hex: 0x120A08)
    static let surface = Color(hex: 0x1B100B)
    static let surfaceVariant = Color(hex: 0x261611)
    static let surfaceHigh = Color(hex: 0x321C14)

    // Accent — Volcanic Arc (flame orange)
    static let primary = Color(hex: 0xFF5A1F)
    static let primaryMuted = Color(hex: 0xB93A13)
    static let primaryGlow = Color(hex: 0xFF5A1F, opacity: 0.24)

    // Accent — Ember amber
    static let secondary = Color(hex: 0xFF9F2E)
    static let secondaryMuted = Color(hex: 0xD97512)
    static let secondaryGlow = Color(hex: 0xFF9F2E, opacity: 0.24)

    // Accent — Hot magenta
    static let tertiary = Color(hex: 0xFF2D78)
    static let tertiaryMuted = Color(hex: 0xB51F5B)

    // Semantic
    static let error = Color(hex: 0xFF5B5B)
    static let success = Color(hex: 0x70D68B)
    static let warning = Color(hex: 0xFFC04A)
    static let info = Color(hex: 0xFF8A3A)

    // Text
    static let textPrimary = Color(hex: 0xFFF1EA)
    static let textSecondary = Color(hex: 0xE0B8A4)
    static let textTertiary = Color(hex: 0x9E6A52)
    static let textOnPrimary = Color(hex: 0x1B100B)

    // Borders & dividers
    static let border = Color(hex: 0x4A2718)
    static let borderSubtle = Color(hex: 0x372013)
    static let divider = Color(hex: 0x261611)
}
